src/main/java/com/epam/test_generator/services/StepService.java

```diff
@@ -113,4 +113,30 @@ public void removeStep(Long suitId, Long caseId, Long stepId) {
         caze.getSteps().remove(step);
         stepDAO.delete(stepId);
     }
+
+    /**
+     * This service method is specialized for 'adding' 'editing' and 'deleting' a list of steps
+     *
+     * For selecting required action uses "action" field from DTO object
+     *
+     * @param steps array list of steps
+     */
+    public void cascadeUpdateSteps(long suitId, long caseId, List<StepDTO> steps) {
+        for (StepDTO stepDTO: steps) {
+            String action = stepDTO.getAction();
+            if(action != null){
+                switch(action.toLowerCase()) {
+                    case "update":
+                        updateStep(suitId, caseId, stepDTO.getId(), stepDTO);
+                        break;
+                    case "create":
+                        addStepToCase(suitId, caseId, stepDTO);
+                        break;
+                    case "delete":
+                        removeStep(suitId, caseId, stepDTO.getId());
```

Comment:
I think that it is not a good idea to update, add or remove steps separately. There are a lot of extra queries to database in this code.

Reply:
You are right, but in the current task, this was not supposed. We should think about it and take it as the new task on the next sprint if it would be necessary.